### PR TITLE
[BEAM-3457] Add Go Gradle precommit

### PR DESCRIPTION
**Pull request by herohde**

### CODE REVIEW

**Comment by herohde** [comment]
run seed job

**Comment by herohde** [comment]
run go gradle precommit

**Comment by herohde** [comment]
run seed job

**Comment by herohde** [comment]
Run Seed Job

File: .test-infra/jenkins/job_beam_PreCommit_Go_GradleBuild.groovy [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import common_job_properties
+
+// This is the Go precommit which runs a gradle build, and the current set
+// of precommit tests.
+job('beam_PreCommit_Go_GradleBuild') {
+  description('Runs an install of the current GitHub Pull Request.')
```
Comment (lukecwik) [inline .test-infra/jenkins/job_beam_PreCommit_Go_GradleBuild.groovy]: This description seems incorrect.

File: build.gradle [hunk 2]
```diff
@@ -298,3 +301,7 @@ task javaPreCommit() {
   }
   dependsOn ":examples:java:preCommit"
 }
+
+task goPreCommit() {
+  dependsOn ":sdks:go:test"
+}
```
Comment (lukecwik) [inline build.gradle]: add missing newline at EOF

File: .test-infra/jenkins/job_beam_PreCommit_Go_GradleBuild.groovy (hunk 1 above)
Comment (herohde) [reply .test-infra/jenkins/job_beam_PreCommit_Go_GradleBuild.groovy]: Yes. I stole it from Java ...

Comment (lukecwik) [reply .test-infra/jenkins/job_beam_PreCommit_Go_GradleBuild.groovy]: That one is wrong too then.

File: build.gradle (hunk 2 above)
Comment (herohde) [reply build.gradle]: Done

File: .test-infra/jenkins/job_beam_PreCommit_Go_GradleBuild.groovy (hunk 1 above)
Comment (herohde) [reply .test-infra/jenkins/job_beam_PreCommit_Go_GradleBuild.groovy]: Fixed both

**Comment by herohde** [comment]
Thanks @lukecwik. PTAL

**Comment by herohde** [comment]
retest this please

**Comment by lukecwik** [comment]
Run Go Gradle PreCommit

**Comment by herohde** [comment]
Run Go Gradle PreCommit

**Comment by lukecwik** [comment]
This seems to be failing on pulling down thrift:
Execution failed for task ':sdks:go:resolveBuildDependencies'.
> Exception in resolution, message is:
  Cannot recognized package: git.apache.org/thrift.git/lib/go/thrift
  Resolution stack is:
  +- github.com/apache/beam/sdks/go
   +- golang.org/x/oauth2#a032972e28060ca4f5644acffae3dfc268cc09db
    +- cloud.google.com/go#c1d6c20e26a9664da077a2bcdac922e4bc2d5766
     +- go.opencensus.io#7a3c63ab1c5772f940fb091832581522533e432f

There is a newer version of the gogradle plugin with some dependency management fixes, consider doing the upgrade as it may resolve some of the issues.

**Comment by herohde** [comment]
Sent out #4619 for the thrift resolution problem.

**Comment by herohde** [comment]
Run Go Gradle PreCommit

**Comment by herohde** [comment]
Run Seed Job

**Comment by lukecwik** [comment]
Run Go Gradle PreCommit

**Comment by herohde** [comment]
Run Go Gradle PreCommit

**Comment by herohde** [comment]
Run Seed Job

**Comment by lukecwik** [comment]
Run Go Gradle PreCommit

**Comment by kennknowles** [comment]
Can this replace the maven Golang precommit? Currently still running.

**Comment by herohde** [comment]
The idea is that it'll replace the maven Go precommit. I just want to run them both for a few weeks before doing do.

**Pull request merged**